### PR TITLE
Monitor resources version bump

### PR DESCRIPTION
**Pull request by DomGarguilo**
This PR updates javascript dependencies used in the accumulo monitor.

* bumps jquery from 3.6.1 -> 3.7.1 (files downloaded from https://jquery.com/download/ )
* bumps DataTables from 1.12.1 -> 1.13.6 (files downloaded from https://www.datatables.net/download/)
* bumps flot from 4.2.3 to 4.2.6 (files downloaded from https://github.com/flot/flot/releases/tag/v4.2.6)
* bumps Bootstrap from 5.0.2 -> 5.3.1 (files downloaded from https://getbootstrap.com/docs/5.3/getting-started/download/)
    * For some reason this upgrade required me to also include a new package, Popper, which is used for the dropdown menu functionality. For some reason, the docs say 5.0.2 also requires it but for some reason the dropdowns were working fine with this version without it.
* Updates the LICENCE files accordingly

### CODE REVIEW

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/default.ftl [hunk 1]
```diff
@@ -30,7 +30,8 @@
         ${val}
       </#list>
     <#else>
-      <script src="/resources/external/jquery/jquery-3.6.1.js"></script>
+      <script src="/resources/external/jquery/jquery-3.7.1.js"></script>
+      <script src="/resources/external/popper/popper.js"></script>
       <script src="/resources/external/bootstrap/js/bootstrap.js"></script>
```
Comment (ctubbsii) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/default.ftl:35]: We don't really need popper separate. I'm not even sure why they made it possible to separate out. It's just used internally in bootstrap, as part of bootstrap.

If you use the bootstrap bundle, the only thing you need to do is update the filename:

```suggestion
      <script src="/resources/external/bootstrap/js/bootstrap.bundle.js"></script>
```

Also, version 5.3.2 is the latest available, as of this comment.

**Review by ctubbsii — CHANGES_REQUESTED** [review]
A few things:

1. Can use bootstrap.bundle.js instead of bootstrap.js to include popper
2. I'm not sure we need to redistribute the .map files for bootstrap; as far as I can tell, they are only used by browsers for debugging bootstrap itself; if users want to add them, they can, but we don't need to distribute them by default and they can be removed from our source, so long as that doesn't cause any errors, which I'm pretty sure it won't.
3. DataTables now uses unicode characters for its up/down arrows for sorting. It is no longer necessary to ship the datatables/images; those can be deleted
4. DataTables does not need the css/jquery.dataTables.css file, since it has the css/dataTables.bootstrap5.css for styling, so that can be deleted too.
5. Bootstrap is now at 5.3.2 as of this review; this PR is for 5.3.1.

**Comment by DomGarguilo** [comment]
> 2\. I'm not sure we need to redistribute the .map files for bootstrap; as far as I can tell, they are only used by browsers for debugging bootstrap itself; if users want to add them, they can, but we don't need to distribute them by default and they can be removed from our source, so long as that doesn't cause any errors, which I'm pretty sure it won't.

I was wondering about this. if we could remove all of the map files since it seems like they are usually used for debugging minified files anyways (which we dont have here).

**Comment by DomGarguilo** [comment]
> A few things:
> 
> 1. Can use bootstrap.bundle.js instead of bootstrap.js to include popper
> 2. I'm not sure we need to redistribute the .map files for bootstrap; as far as I can tell, they are only used by browsers for debugging bootstrap itself; if users want to add them, they can, but we don't need to distribute them by default and they can be removed from our source, so long as that doesn't cause any errors, which I'm pretty sure it won't.
> 3. DataTables now uses unicode characters for its up/down arrows for sorting. It is no longer necessary to ship the datatables/images; those can be deleted
> 4. DataTables does not need the css/jquery.dataTables.css file, since it has the css/dataTables.bootstrap5.css for styling, so that can be deleted too.
> 5. Bootstrap is now at 5.3.2 as of this review; this PR is for 5.3.1.

Took care of all of this in 983b1f3.

**Comment by DomGarguilo** [comment]
https://github.com/apache/accumulo/blob/983b1f3a8a37980a5ed0d5c43d9c367efe290205/server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/default.ftl#L39-L46

One thing I was wondering is if we need to distribute the files that are not used here. There are many other flot files that are not used but we still have in the codebase. I tried removing all of the files (in 6bc081a) that don't appear in the script tags above and everything seems to be working fine.

**Comment by ctubbsii** [comment]
> Took care of all of this in [983b1f3](https://github.com/apache/accumulo/commit/983b1f3a8a37980a5ed0d5c43d9c367efe290205).

I think you left the changes to the LICENSE files from adding popper. Those aren't needed, since they are part of the bootstrap bundle. And this also kept the bootstrap at 5.3.1 instead of the current 5.3.2

**Comment by ctubbsii** [comment]
> One thing I was wondering is if we need to distribute the files that are not used here. There are many other flot files that are not used but we still have in the codebase. I tried removing all of the files (in [6bc081a](https://github.com/apache/accumulo/commit/6bc081a4e5a04d408e0a7537719a821d4a8867b7)) that don't appear in the script tags above and everything seems to be working fine.

The flot distribution contains all of them. I don't care too much if we only pick a subset of the ones we use, but it might be hard to maintain if we're constantly having to trim down the distribution to just those we use, each time we update. It's up to you.

**Comment by DomGarguilo** [comment]
> > One thing I was wondering is if we need to distribute the files that are not used here. There are many other flot files that are not used but we still have in the codebase. I tried removing all of the files (in [6bc081a](https://github.com/apache/accumulo/commit/6bc081a4e5a04d408e0a7537719a821d4a8867b7)) that don't appear in the script tags above and everything seems to be working fine.
> 
> The flot distribution contains all of them. I don't care too much if we only pick a subset of the ones we use, but it might be hard to maintain if we're constantly having to trim down the distribution to just those we use, each time we update. It's up to you.

The bootstrap and DataTables distributions contain many files which we cherry pick from as well so I was thinking it would be okay to do the same with flot. I ended up reverting that change though so all of the files are still there. Also I see in 2262125 you removed popper from the LICENSE files which i forgot to do, thanks. Do you plan do push a commit to fix the bootstrap version in the LICENSE files as well or should I go ahead and do that?

**Comment by ctubbsii** [comment]
@DomGarguilo wrote:
> The bootstrap and DataTables distributions contain many files which we cherry pick from as well so I was thinking it would be okay to do the same with flot.

True. I guess the way I was downloading them made it feel different than cherry-picking what we needed.

For flot, the way I was getting it was:

1. Download the zip for the latest tag in GitHub: https://github.com/flot/flot/tags
2. Extract the entire package/source directory and the package/LICENSE.txt file from the distribution tarball inside that zip
3. Remove the redundant jquery that we don't need

So, I'm basically keeping it the way I got it, although it took a few steps to get that far. It didn't seem like it was worth doing more work to trim it down from there.

For DataTables and Bootstrap, it was easier, because they basically have download links on their sites to just the files you need, without downloading the full zip and picking stuff out of it.

DataTables:
1. Went to https://datatables.net/download/
2. Selected Bootstrap 5 for the styling framework, and unchecked the minify and concatenate options
3. Downloaded all 3 files from the CDN into the project, exactly as they are

Bootstrap:
1. Went to https://getbootstrap.com/
2. Scrolled down to the CDN links where only two files are listed: bootstrap.bundle.min.js and bootstrap.min.css
3. Removed ".min" from the link URLs, and downloaded all files directly into the project, exactly as they are
  

> Do you plan do push a commit to fix the bootstrap version in the LICENSE files as well or should I go ahead and do that?

We can either fix the LICENSE files to match the files in the project, or we can just update the files in the project to the newer version specified in the LICENSE. I think updating to the newer version is better. If you don't get to it, I can get to it eventually. (Part of the reason I detailed all the steps I took above, is to make it easier on whoever ends up doing it.)

**Comment by DomGarguilo** [comment]
> EDIT: I must've made a mistake. I just checked, and the files are definitely the 5.3.2 version, which matches the commit message and the LICENSE files. I'm not sure what I saw before, but this doesn't look like a problem right now, so I must've made a mistake before.

I see 5.3.1 in the LICENSE files:
https://github.com/apache/accumulo/blob/51124ad34872c2b659b69251fcd1160e20bbdd44/LICENSE#L316

**Pull request merged**